examples/firebase-auth-firestore/app/routes/login.tsx

```diff
@@ -44,12 +70,38 @@ export const action: ActionFunction = async ({ request }) => {
 };
 
 export default function Login() {
+  const [clientAction, setClientAction] = useState<ActionData>();
   const action = useActionData<ActionData>();
+  const restConfig = useLoaderData<LoaderData>();
+  const submit = useSubmit();
+
+  const handleSubmit = useCallback(
+    async (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
```

Comment:
this line is my main issue with this approach, as if something fails after this point it will not fall back to the form post + server side auth